### PR TITLE
feat: Support workflows.ts as default workflowsPath

### PR DESCRIPTION
**Pull request by joebowbeer**
## What was changed

**_[DRAFT]_** workflowsPath defaults to `workflows.ts` if `workflows` path does not exist.

## Why?

Typescript programmers will expect to see Node-style import resolution.

Relates to #258

## Checklist

2. How was this tested:

This passes npm test including integration tests.

HOWEVER, the intended behavior is not covered by any of the test cases.

### CODE REVIEW

File: packages/worker/src/worker.ts [hunk 1]
```diff
@@ -272,11 +273,17 @@ function includesDeps<T extends WorkerSpec>(
 
 export function addDefaults<T extends WorkerSpec>(options: WorkerSpecOptions<T>): WorkerOptionsWithDefaults<T> {
   const { workDir, ...rest } = options;
+  // If workDir is defined, workflowsPath defaults to workflows.js if workflows does not exist
+  let workflowsPath = undefined;
+  if (workDir) {
+    const workflowsDir = resolve(workDir, 'workflows');
+    workflowsPath = existsSync(workflowsDir) ? workflowsDir : resolve(workDir, 'workflows.js');
```
Comment (bergundy) [inline packages/worker/src/worker.ts:280]: This is incorrect, a user might have both `workflows/workflowA.ts` and `workflows.ts`.
We also support reading either `js` or `ts` extensions (for better source maps, default when running with `ts-node`).

If you address all of these issues along with handling interceptors loading when `workflowsPath` is a file than we can merge this.

**Review by bergundy — CHANGES_REQUESTED** [review]
Thanks @joebowbeer for yet another contribution!

File: packages/worker/src/worker.ts (hunk 1 above)
Comment (joebowbeer) [reply packages/worker/src/worker.ts:280]: Thanks. To be rigorously correct, though, don't these checks need to be performed at runtime? I don't think watch is smart enough to know that adding or removing a workflows file/folder will invalidate this logic.

I wish I knew how this path is loaded at runtime. For example, where is the code that loads index.js now?

Comment (bergundy) [reply packages/worker/src/worker.ts:280]: What do you mean by "at runtime"?
I watch mode, nodemon will restart the process anytime a file changes and this logic will be re-evaluated.
`index.{j,t}s` is loaded by webpack in `isolate-builder.ts`.

Comment (joebowbeer) [reply packages/worker/src/worker.ts:280]: I see. I suppose this needs to work with any custom loader, e.g., babel-loader as well as ts-loader.

**Comment by lorensr** [comment]
I think we'll be removing `workDir` (#269). Is there anything we can default `workflowsPath` to without that? Eg `workflows.ts` in the same directory as worker?

**Comment by joebowbeer** [comment]
The list of extensions is dependent on the loader, e.g., ts-loader

The webpack module resolver or custom loader should be able to work with just "workflows"

**Pull request closed without merging**